### PR TITLE
Update to JupyterLab 3.4

### LINKED ISSUE #2463: Elyra R editor vs JupyterLab R editor
I've noticed that a new R editor tile is displayed in the launcher:

image

Quo Vadis?

### PR DESCRIPTION
**Pull request by ajbozarth**
Update to Jupyterlab 3.4 and fix launcher issues blocked by the 3.4 release:
- Addresses the duplicate launcher cards for the python and r files https://github.com/jupyterlab/jupyterlab/pull/12344
- Updates to include support for the add tab feature https://github.com/jupyterlab/jupyterlab/pull/12195

Fixes #2463

Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: packages/theme/src/launcher.tsx [hunk 1]
```diff
@@ -32,6 +33,51 @@ import * as React from 'react';
...
+  items(): IIterator<ILauncher.IItemOptions> {
+    const items: ILauncher.IItemOptions[] = [];
+
+    let pyEditorInstalled = false;
+    let rEditorInstalled = false;
+
+    this._items.forEach(item => {
+      if (item.command === CommandIDs.createNewPythonEditor) {
+        pyEditorInstalled = true;
+      } else if (item.command === CommandIDs.createNewPythonEditor) {
```
Comment (karlaspuldaro) [inline packages/theme/src/launcher.tsx:55]:
```suggestion
      } else if (item.command === CommandIDs.createNewREditor) {
```

Comment (ajbozarth) [reply packages/theme/src/launcher.tsx]: thanks for the typo catch, fixed

**Comment by ptitzler** [comment]
The upstream PR https://github.com/jupyterlab/jupyterlab/pull/12344 was merged and is expected to be included in JupyterLab 3.4. Adding label to this PR.

**Comment by ptitzler** [comment]
Assigning 3.8 milestone but might have to defer if JupyterLab 3.4 is not available before we cut that release.

**Comment by ajbozarth** [comment]
Pending 3.4 I've push a couple untested updates including fixing the var name (I changed it upstream prior to committing) and adding support for https://github.com/jupyterlab/jupyterlab/pull/12195 since that requires edits to the same file

**Review by karlaspuldaro — APPROVED** [review]
Looks good! Thank you @ajbozarth :)

**Review by ptitzler — APPROVED** [review]
Thank you! LGTM!


Tiles:

image

Link to pipeline editor settings from empty canvas:

image

**Review by kiersten-stokes — APPROVED** [review]
LGTM!

**Review by akchinSTC — APPROVED** [review]

**Pull request merged**